pkg/api/norman/customization/aks/listers.go

```diff
@@ -158,7 +158,17 @@ func listKubernetesUpgradeVersions(ctx context.Context, clusterLister mgmtv3.Clu
 		return nil, http.StatusBadRequest, fmt.Errorf("invalid cluster id")
 	}
 
-	resp.CurrentVersion = *cluster.Spec.AKSConfig.KubernetesVersion
+	if cluster.Spec.AKSConfig.KubernetesVersion != nil {
+		resp.CurrentVersion = *cluster.Spec.AKSConfig.KubernetesVersion
+	} else {
+		if cluster.Status.AKSStatus.UpstreamSpec == nil {
+			return nil, http.StatusBadRequest, fmt.Errorf("no upstream spec found")
+		}
+		if cluster.Status.AKSStatus.UpstreamSpec.KubernetesVersion == nil {
+			return nil, http.StatusBadRequest, fmt.Errorf("no kubernetes version found in upstream spec")
+		}
```

Comment:
The user doesn't know that the upstreamSpec exists, so telling them that it (or something on it) doesn't exist won't make sense to them.

I think something like "kubernetes version of the cluster cannot be determined" in both cases is better.